### PR TITLE
CORE-240: Re-Propagate updated public key

### PR DESCRIPTION
**Pull request by leviathanch**
The most significant change in this patch is that PEP_ct_mistrusted is now being upgraded and handled like PEP_ct_key_expired in validate_fpr. This change is nessecary so that a prior but reset identity can be resumed from the database in order to communicate the update of the key using which needs to be signed with the old key for security reasons.

### CODE REVIEW

File: src/key_reset.c [hunk 1]
```diff
@@ -710,8 +710,9 @@ PEP_STATUS receive_key_reset(PEP_SESSION session,
 
         // Basically, see if fpr is even in the database
         // for this user - we'll get PEP_ct_unknown if it isn't
-        if (ct_result == PEP_ct_unknown)
+        if (ct_result == PEP_ct_unknown) {
```
Comment (ghost) [inline src/key_reset.c:713]: YES!

File: src/key_reset.c [hunk 2]
```diff
@@ -901,15 +904,22 @@ PEP_STATUS create_standalone_key_reset_message(PEP_SESSION session,
...
     
     message* reset_msg = NULL;
-    
-    PEP_STATUS status = _generate_keyreset_command_message(session, own_identity,
-                                                           recip,
-                                                           old_fpr, new_fpr, false,
-                                                           &reset_msg);
-                            
+
+    own_identity = new_identity(own_identity->address, old_fpr, PEP_OWN_USERID, own_identity->username);
```
Comment (ghost) [inline src/key_reset.c:914]: Memory got leaked

File: src/key_reset.c [hunk 3]
```diff
@@ -901,15 +904,22 @@ PEP_STATUS create_standalone_key_reset_message(PEP_SESSION session,
...
+
+    status = set_own_key(session, own_identity, old_fpr);
```
Comment (ghost) [inline src/key_reset.c:916]: This writes to the DB and triggers certain sync events. Can we avoid it?

File: src/keymanagement.c [hunk 4]
```diff
@@ -200,6 +201,7 @@ PEP_STATUS validate_fpr(PEP_SESSION session,
             // delete key from being default key for all users/identities
             status = remove_fpr_as_default(session, fpr);
             // fallthrough intentional!
+        case PEP_ct_mistrusted:
```
Comment (ghost) [inline src/keymanagement.c:204]: This function is used all over the place, and now has changed behavior, including DB writes. Why is this needed?

File: src/message_api.c [hunk 5]
```diff
@@ -6268,23 +6268,14 @@ static PEP_STATUS _decrypt_message(
...
-                                    goto pEp_error; // out of memory
-
-                                identity_list* used_cc_ids = NULL;    
-                                status = ident_list_intersect(my_rev_ids, msg->cc, &used_cc_ids);
-                                if (status != PEP_STATUS_OK)
-                                    goto pEp_error;
-
-                                used_ids_for_key = identity_list_join(used_ids_for_key, used_cc_ids);
-                                
+                                used_ids_for_key = identity_list_add(used_ids_for_key, msg->from);
```
Comment (ghost) [inline src/message_api.c:6271]: You are checking for messages from you to yourself that used an outdated key?

File: src/message_api.c [hunk 6]
```diff
@@ -6268,23 +6268,14 @@ static PEP_STATUS _decrypt_message(
                             if (status == PEP_STATUS_OK && my_rev_ids) {
                                 // get identities in this list the message was to/cc'd to (not for bcc)
                                 identity_list* used_ids_for_key = NULL;
-                                status = ident_list_intersect(my_rev_ids, msg->to, &used_ids_for_key);
-                                if (status != PEP_STATUS_OK)
-                                    goto pEp_error; // out of memory
-
-                                identity_list* used_cc_ids = NULL;    
-                                status = ident_list_intersect(my_rev_ids, msg->cc, &used_cc_ids);
```
Comment (ghost) [inline src/message_api.c:6276]: Why did the CC disappear?

**Review by ghost — CHANGES_REQUESTED** [review]

File: src/keymanagement.c (hunk 4 above)
Comment (leviathanch) [reply src/keymanagement.c:204]: You are right, I just made a make clean and make test and now some tests fail.
I think I need to rethink those changes and do this differently.
It was needed in order to set the key for signing the mail with an old identity, but I think I'll have to do this differently.

File: src/message_api.c (hunk 5 above)
Comment (leviathanch) [reply src/message_api.c:6271]: No, from is basically where the mail comes from, isn't it?

File: src/message_api.c (hunk 6 above)
Comment (leviathanch) [reply src/message_api.c:6276]: Ooops, good point! I'll fix that too.

File: src/key_reset.c (hunk 1 above)
Comment (saschabacardit) [reply src/key_reset.c:713]: I personally prefer the inline version :^)
But if the main two ([leviathanch](https://github.com/leviathanch) & [planckdz](https://github.com/planckdz)) prefer to not use it, we should note it down somewhere along with the proper set up (iirc we did discuss it and end up on the standard being the Gnu one for grammar?).

Comment (ghost) [reply src/key_reset.c:713]: Yes, you are right of course.

* Keep pure formatting changes to a minimum outside the real code changes.
* Agree on a clang format file and call it a day. There is already one (`.clang-format`), please give feedback.

I can certainly live with the "braceless if" style, but it _will_ trip you up sooner or later.

Comment (leviathanch) [reply src/key_reset.c:713]: It's more a leftover from me debugging with printfs xD

File: src/key_reset.c (hunk 2 above)
Comment (leviathanch) [reply src/key_reset.c:914]: Yikes

File: src/key_reset.c (hunk 3 above)
Comment (leviathanch) [reply src/key_reset.c:916]: Yup. Is avoidable, fixed in PR2 which is following soon
